ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/hardware_counters.schema.yml

```diff
@@ -0,0 +1,131 @@
+      filename: data_model/Hardware
+    type: dict
+    keys:
+      features:
+        type: list
+        description: |
+          This data model allows to configure the list of hardware counters feature
+          available on Arista platforms.
+
+          The `name` key accepts a list of valid_values which MUST be updated to support
+          new feature as they are released in EOS.
+
+          The available values of the different keys like 'direction' or 'address_type'
+          are feature and hardware dependent and this model DOES NOT validate that the
+          combinations are valid. It is the responsability of the user of this data model
+          to make sure that the rendered CLI is accepted by the targeted device.
+
+          Examples:
+
+            * Use:
+              ```yaml
+              hardware_counters:
+                features:
+                  - name: ip
+                    direction: out
+                    layer3: true
+                    units_packets: true
+              ```
+
+              to render:
+              ```eos
+              hardware counter feature ip out layer3 units packets
+              ```
+            * Use:
+              ```yaml
+              hardware_counters:
+                features:
+                  - name: route
+                    address_type: ipv4
+                    vrf: test
+                    prefix: 192.168.0.0/24
+              ```
+
+              to render:
+              ```eos
+              hardware counter feature route ipv4 vrf test 192.168.0.0/24
+              ```
+
+        # primary_key: name
+        # no primary key to allow multiple times the same name
+        items:
+          type: dict
+          keys:
+            name:
+              type: str
+              valid_values:
+                - acl
+                - decap-group
+                - directflow
+                - ecn
+                - flow-spec
+                - gre tunnel interface
+                - ip
+                - mpls interface
+                - mpls lfib
+                - mpls tunnel
+                - multicast
+                - nexthop
+                - pbr
+                - pdp
+                - policing interface
+                - qos
+                - qos dual-rate-policer
+                - route
+                - routed-port
+                - subinterface
+                - tapagg
+                - traffic-class
+                - traffic-policy
+                - vlan
+                - vlan-interface
+                - vni decap
+                - vni encap
+                - vtep decap
+                - vtep encap
+            direction:
+              type: str
+              valid_values:
+                - in
+                - out
+                - cpu
+              description: |
+                Most features support only 'in' and 'out'. Some like traffic-policy support 'cpu'.
+                Some features DO NOT have any direction.
```

Comment:
```suggestion
                Some features do not have any direction.
```

Reply:
Using $ref to point to the `eos_cli_config_gen` schema.